core/src/test/scala/org/apache/spark/sql/delta/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -0,0 +1,311 @@
+
+      df2.write.format("delta").mode("append").saveAsTable(noStatsTableName)
+    }
+
+    withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "true") {
+      import io.delta.tables._
+
+      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(testTableName)
+      df.write.format("delta").mode(SaveMode.Overwrite).save(testTablePath)
+
+      spark.sql(s"DELETE FROM $testTableName WHERE id = 1")
+      DeltaTable.forPath(spark, testTablePath).delete("id = 1")
+      spark.sql(s"DELETE FROM $mixedStatsTableName WHERE id = 1")
+
+      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(testTableName)
+      df2.write.format("delta").mode(SaveMode.Append).save(testTablePath)
+      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(mixedStatsTableName)
+    }
+  }
+
+  test("Select Count: basic") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT COUNT(*) FROM $testTableName",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: column alias") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT COUNT(*) as MyColumn FROM $testTableName",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: table alias") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT COUNT(*) FROM $testTableName MyTable",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: time travel") {
+    checkResultsAndOptimizedPlan(s"SELECT COUNT(*) FROM $testTableName VERSION AS OF 0",
+      Seq(Row(3L)),
+      "LocalRelation [none#0L]")
+
+    checkResultsAndOptimizedPlan(s"SELECT COUNT(*) FROM $testTableName VERSION AS OF 1",
+      Seq(Row(2L)),
+      "LocalRelation [none#0L]")
+
+    checkResultsAndOptimizedPlan(s"SELECT COUNT(*) FROM $testTableName VERSION AS OF 2",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: external") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT COUNT(*) FROM delta.`$testTablePath`",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: sub-query") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT (SELECT COUNT(*) FROM $testTableName)",
+      Seq(Row(totalRows)),
+      "Project [scalar-subquery#0 [] AS #0L]\n:  +- LocalRelation [none#0L]\n+- OneRowRelation")
+  }
+
+  test("Select Count: as sub-query filter") {
+    checkResultsAndOptimizedPlan(
+      s"SELECT 'ABC' WHERE (SELECT COUNT(*) FROM $testTableName) = $totalRows",
+      Seq(Row("ABC")),
+      "Project [ABC AS #0]\n+- Filter (scalar-subquery#0 [] = " +
+        totalRows + ")\n   :  +- LocalRelation [none#0L]\n   +- OneRowRelation")
+  }
+
+  test("Select Count: limit") {
+    // Limit doesn't affect COUNT results
+    checkResultsAndOptimizedPlan(
+      s"SELECT COUNT(*) FROM $testTableName LIMIT 3",
+      Seq(Row(totalRows)),
+      "LocalRelation [none#0L]")
+  }
+
+  test("Select Count: snapshot isolation") {

```

Comment:
This test will always take 1 minute. Could you try to use `PrepareDeltaScanBase.withCallbackOnGetDeltaScanGenerator` to write a faster test instead? See https://github.com/delta-io/delta/blob/3a8f84330df9457095ba6cfb34e18d23e341d261/core/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeGeneratedColumnSuite.scala#L1252 as an example.

Reply:
Done, thanks.